ingest.py

```diff
@@ -83,60 +83,89 @@ def load(self) -> List[Document]:
 
 def load_single_document(file_path: str) -> Document:
     ext = "." + file_path.rsplit(".", 1)[-1]
-    if ext in LOADER_MAPPING:
-        loader_class, loader_args = LOADER_MAPPING[ext]
-        loader = loader_class(file_path, **loader_args)
-        return loader.load()[0]
 
-    raise ValueError(f"Unsupported file extension '{ext}'")
+    loader_class, loader_args = LOADER_MAPPING[ext]
+    loader = loader_class(file_path, **loader_args)
+    return loader.load()[0]
 
 
-def load_documents(source_dir: str, ignored_files: List[str] = []) -> List[Document]:
+def load_documents(
+    source_dir: str,
+    ignored_files: Optional[List[str]] = None,
+    report_processed_files: bool = True,
+    report_skipped_files: bool = True,
+    report_ignored_files: bool = True,
+    use_process_bar: bool = True,
+) -> List[Document]:
+
     """
     Loads all documents from the source documents directory, ignoring specified files
     """
-    all_files = []
-    for ext in LOADER_MAPPING:
-        all_files.extend(
-            glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
-        )
-    filtered_files = [file_path for file_path in all_files if file_path not in ignored_files]
+    if ignored_files is None:
+        ignored_files : List[str] = []
+
+    filtered_files = []
+    extensions = tuple(LOADER_MAPPING.keys())
+
+    for root, _, files in os.walk(source_dir):
+        for file in files:
+            if file.endswith(extensions):
+                file_path = os.path.join(root, file)
+                if file_path not in ignored_files:
+                    filtered_files.append(file_path)
+                else:
+                    if report_ignored_files:
+                        print(f"Ignored '{file_path}' (ignore list)")
+            else:
+                if report_skipped_files:
+                    file_path = os.path.join(root, file)
+                    print(f"Skipping '{file_path}' - unmatched extension")
 
     with Pool(processes=os.cpu_count()) as pool:
         results = []
-        with tqdm(total=len(filtered_files), desc='Loading new documents', ncols=80) as pbar:
+        with (tqdm(total=len(filtered_files), desc='Loading new documents', ncols=80) if use_process_bar else None) as pbar:
             for i, doc in enumerate(pool.imap_unordered(load_single_document, filtered_files)):
                 results.append(doc)
-                pbar.update()
+                if use_process_bar:
+                    pbar.update()
+                if report_processed_files:
+                    file_path = filtered_files[i]
+                    print(f"Processed '{file_path}'")
```

Comment:
I think the printing inside the `tqdm` is creating some problems.
I see this:
```sh
Skipping 'source_documents/state_of_the_union.tx' - unmatched extension
Loading new documents:   0%|                              | 0/4 [00:00<?, ?it/s]Processed 'source_documents/bar.csv'
Processed 'source_documents/foo.txt'
Processed 'source_documents/A/barfoo.csv'
Processed 'source_documents/B/foobar.csv'
Loading new documents: 100%|█████████████████████| 4/4 [00:00<00:00, 584.88it/s]
```

Reply:
This is an expected "misbehavior".

Which is why these are options, there should CLI argument parsing that disables the progress bar when file reporting is active, unless the user forces the progress bar.

As indicated above, the CLI argument parsing is left to another volunteer.

Still, with the progress bar the user can still see how much progress there is.

Reply:
I think using `tqdm.write(f"Processed '{file_path}'")` instead of `print()` should fix it.

Reply:
Pushed the change.